miden-lib/asm/sat/internal/asset_vault.masm

```diff
@@ -92,32 +88,33 @@ export.add_fungible_asset
     dup.4 movdn.5 push.0 movdn.3 dup movdn.4
     # => [ASSET_KEY, faucet_id, amount, vault_root_ptr, vault_root_ptr]
```

Comment:
nit: we could probably avoid making a duplicate of `vault_root_ptr` here because we can always just use `dup.10` instead of `movup.10` on line 92 below.

Reply:
good catch.